search-resources-api/src/main/java/no/unit/nva/search/SearchTicketsHandler.java

```diff
@@ -18,156 +30,162 @@
-        String customerCristinInstitutionNumber = extractInstitutionNumberFromRequestedOrganization(customerCristinId);
-        return customerCristinInstitutionNumber.equals(requestedOrgInstitutionNumber);
-    }
-
-    private String extractInstitutionNumberFromRequestedOrganization(URI requestedOrg) {
-        String requestedOrgCristinIdentifier = UriWrapper.fromUri(requestedOrg).getLastPathElement();
-        return requestedOrgCristinIdentifier.split(CRISTIN_ORG_LEVEL_DELIMITER)[HIGHEST_LEVEL_ORGANIZATION];
-    }
-
-    private String getIndexName(RequestInfo requestInfo) {
-        String requestPath = RequestUtil.getRequestPath(requestInfo);
-        return UnixPath.of(requestPath).getLastPathElement();
-    }
+  public static final String VIEWING_SCOPE_QUERY_PARAMETER = "viewingScope";
+  public static final String CRISTIN_ORG_LEVEL_DELIMITER = "\\.";
+  public static final int HIGHEST_LEVEL_ORGANIZATION = 0;
+  public static final String EXPECTED_ACCESS_RIGHT_FOR_VIEWING_MESSAGES_AND_DOI_REQUESTS = "APPROVE_DOI_REQUEST";
+  private static final Logger logger = LoggerFactory.getLogger(SearchTicketsHandler.class);
+  public static final String ROLE_CURATOR = "curator";
+  public static final String PARAM_ROLE = "role";
+  private final SearchClient searchClient;
+  private final IdentityClient identityClient;
+
+  @JacocoGenerated
+  public SearchTicketsHandler() {
+    this(new Environment(), defaultSearchClient(), defaultIdentityClient());
+  }
+
+  public SearchTicketsHandler(Environment environment, SearchClient searchClient, IdentityClient identityClient) {
+    super(Void.class, environment, objectMapperWithEmpty);
+    this.searchClient = searchClient;
+    this.identityClient = identityClient;
+  }
+
+  @Override
+  protected SearchResponseDto processInput(Void input, RequestInfo requestInfo, Context context)
+      throws ApiGatewayException {
+
+    var indexName = getIndexName(requestInfo);
+    logger.info("Index name: {}", indexName);
+    assertUserHasAppropriateAccessRights(requestInfo);
+    var role = requestInfo.getQueryParameterOpt(PARAM_ROLE).orElse(ROLE_CURATOR);
+    if (ROLE_CURATOR.equals(role)) {
+      return handleCuratorSearch(requestInfo, indexName);
+    } else {
+      return handleCreatorSearch(requestInfo, indexName);
+    }
+  }
+
+  private SearchResponseDto handleCreatorSearch(RequestInfo requestInfo, String indexName)
+      throws UnauthorizedException, BadGatewayException {
+    final var owner = requestInfo.getUserName();
+    logger.info("OwnerScope: {}", owner);
+    var unread = new FilterAggregationBuilder(
+        "unread",
+        notInFilter(requestInfo, owner, "owner", "viewedBy"));
+    var aggregations = List.of(
+        TYPE_TERMS_AGGREGATION, STATUS_TERMS_AGGREGATION, unread);
+    return searchClient.searchOwnerTickets(toQueryTickets(requestInfo, aggregations), owner, indexName);
+  }
+
```

Comment:
Looks good. But we may need to test this in lambda handler?

Reply:
It's a tricky one. The other 2 are "agnostic" in the sense that if the response contains a certain field, create a bucket aggregation on it. 
This filter bucket need the same/similar query to the main query.
Ideally this would be covered with the tests running with TestContainer in the commons module, but since they requires a knowledge of the query and the values from the request it fits here.